R/DualSimplexSolver.R

```diff
@@ -163,11 +168,11 @@ DualSimplexSolver <- R6Class(
         stop("The data matrix should not contain all zero columns. Use remove_zero_cols() method")
       first_set <-  is.null(self$st$data)
       private$reset_since("data")
-      if (!inherits(data, "ExpressionSet"))
-        data <- create_eset(data)
+      if (!inherits(data, "ExpressionSet")) data <- create_eset(data)
       self$st$data <- add_default_anno(data, gene_anno_lists, sample_anno_lists)
       self$st$scaling <- sinkhorn_scale(exprs(self$st$data), max_iter = sinkhorn_iterations)
-      self$st$proj_full <- svd_project(self$st$scaling, dims = NULL)
+      self$st$proj_ops <- calc_svd_ops(self$get_V_row(), max_dim = max_dim, tol = tol)
+      self$st$proj <- efficient_svd_project(self$get_V_row(), self$get_V_column(), dims = NULL, ops = self$st$proj_ops)
```

Comment:
We don't need it because there is no way we will not call project after this. we only need Sigma values actually.